tiatoolbox/dataloader/wsireader.py

```diff
@@ -84,15 +85,148 @@ def slide_info(self):
         """
         raise NotImplementedError
 
-    def read_region(self, start_w, start_h, end_w, end_h, level=0):
+    def relative_level_scales(self, target_scale, units):
+        """
+        Calculate scale of each image pyramid level relative to the
+        given target scale and units.
+
+        Values > 1 indicate that the level has a larger scale than the
+        target and < 1 indicates that it is smaller.
+
+        Args:
+            target_scale (float): Scale to calculate relative to
+            units (str): Units of the scale. Allowed values are: mpp,
+                power, level, baseline. Baseline refers to the largest
+                resolution in the WSI (level 0).
+
+        Raises:
+            ValueError: Missing MPP metadata
+            ValueError: Missing objective power metadata
+            ValueError: Invalid units
+
+        Returns:
+            list: Scale for each level relative to the given scale and
+                units
+
+        Examples:
+            >>> from tiatoolbox.dataloader import wsireader
+            >>> wsi = wsireader.WSIReader("CMU-1.ndpi")
```

Comment:
I guess this would work with OpenSlideWSIReader not WSIReader

Reply:
This same function should work for both, this docstring example just had a ndpi image in it. I think I have replaced it with a generic path in the new docstring to avoid making it seem like it is ndpi specific.